buildtools/generate_translation_keys.js

```diff
@@ -65,17 +64,31 @@ const getKeysFromFile = (filePath, fileName) => {
   return keys;
 };
 
+// It is highly desirable to retain existing order, to not generate
+// unnecessary merges/conflicts, so we do a specialized merge.
+function merge(target, scanned) {
+  for (const key of Object.keys(scanned)) {
+    if (!(key in target)) {
+      target[key] = scanned[key];
+    } else if (typeof target[key] === 'object') {
+      merge(target[key], scanned[key]);
+    } else {
+      scanned[key] = target[key];
```

Comment:
This is wrong :) and when I fix it, I notice there's an issue around translation of single and multiple versions that I need to look into. Might not be possible to automatically merge, as is.